File: 2015/2015.py
# -*- coding: utf-8 -*-

def get(v):
    sum = 0
    for var in v:
        sum += var
    a = [0] * (sum + 1)
    for i in range(len(v)):
        sum = 0
        for j in range(i, len(v)):
            sum += v[j]
            a[sum] += 1
    return a

def solve(h, w):
    a, b = get(h), get(w)
    result = 0
    for var1, var2 in zip(a, b):
        result += var1 * var2
    return result

if __name__ == "__main__":
    while True:
        (n, m) = map(int, input().split())
        if n == 0 and m == 0:
            break
        h = [int(input()) for i in range(n)]
        w = [int(input()) for i in range(m)]
        print(solve(h, w))

File: 2015/2015.cpp
#include <iostream>
#include <vector>
#define FOR(i, a, n) for(int i = a; i < n; i++)
#define REP(i, n) FOR(i, 0, n)

using namespace std;

vector<int> get(vector<int>& v) {
  int sum = 0;
  REP(i, v.size()) sum += v[i];
  vector<int> a(sum + 1, 0);
  REP(i, v.size()) {
    int sum = 0;
    FOR(j, i, v.size()) a[(sum += v[j])]++;
  }
  return a;
}

int solve(vector<int>& h, vector<int>& w) {
  vector<int> a = get(h), b = get(w);
  int result = 0;
  REP(i, min(a.size(), b.size())) result += a[i] * b[i];
  return result;
}

int main() {
  int n, m;
  while(cin >> n >> m, n || m) {
    vector<int> h(n), w(m);
    REP(i, n) cin >> h[i];
    REP(i, m) cin >> w[i];
    cout << solve(h, w) << endl;
  }
  return 0;
}
